File: TCPIP网络编程/ch02/tcp_server.cc
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#include <arpa/inet.h>
#include <sys/socket.h>
#include <iostream>
void error_handling(char *message);
using namespace std;
int main(int argc, char *argv[])
{
    int serv_sock;
    int client_sock;

    struct sockaddr_in serv_addr;
    struct sockaddr_in client_addr;

    socklen_t client_addr_len;

    char mesg[] = "hello world!";
    if (argc != 2)
    {
        printf("Usage : %s <port> \n", argv[0]);
    }

    // 调用 socket 函数创建套接字
    serv_sock = socket(PF_INET, SOCK_STREAM, 0);
    if (serv_sock == -1)
        error_handling("socket() error");

    memset(&serv_addr, 0, sizeof(serv_addr));

    serv_addr.sin_family = AF_INET;
    serv_addr.sin_addr.s_addr = htonl(INADDR_ANY);
    serv_addr.sin_port = htons(atoi(argv[1]));

    if (bind(serv_sock, (struct sockaddr *)&serv_addr, sizeof(serv_addr)) == -1)
        error_handling("bind() error");

    if (listen(serv_sock, 5) == -1)
        error_handling("listen() error");

    client_addr_len = sizeof(client_addr);
    client_sock = accept(serv_sock, (struct sockaddr *)&client_addr, &client_addr_len);
    if (client_sock == -1)
        error_handling("accept() error!");

    write(client_sock, mesg, sizeof(mesg));
    close(client_sock);
    close(serv_sock);
    return 0;
}

void error_handling(char *message)
{
    fputs(message, stderr);
    fputc('\n', stderr);
    exit(1);
}

File: python/pythonProject/TcpServer.py
import socket

def tcp_server(host="127.0.0.1", port=12345):
    server_socket = socket.socket(socket.AF_INET,socket.SOCK_STREAM)
    server_socket.bind((host,port))
    server_socket.listen(5)
    print(f"[*] 服务器启动, 监听 {host}:{port}")

    while True:
        client_socket, addr = server_socket.accept() # return sock 和 addr
        print(f"[+] 连接来自 {addr}")

        while True:
            data = client_socket.recv(1024)
            if not data:
                break

            print(f"[客户端] {data.decode()}")
            client_socket.sendall(b"Message received")

        client_socket.close()
        print(f"[-] 连接断开 {addr}")

if __name__ == "__main__":
    tcp_server()